### PR TITLE
nmh 1.6 (new formula)

### PR DESCRIPTION
**Pull request by kenh**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

The modern version of the classic RAND MH Mail Handling system.

### CODE REVIEW

File: Formula/nmh.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class Nmh < Formula
+  desc "Nmh, the new version of the MH mail handler"
```
Comment (bfontaine) [inline Formula/nmh.rb]: Please don’t include the formula name in its description.

File: Formula/nmh.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}", "--libdir=#{libexec}",
+                          "--with-cyrus-sasl",
+                          "--with-tls"
+    system "make", "install"
+  end
+
+  test do
+    Dir.mkdir(testpath/"Mail")
+    Dir.mkdir(testpath/"Mail/inbox")
```
Comment (bfontaine) [inline Formula/nmh.rb]: You can remove these two lines; `(testpath/"Mail/inbox/1").write` below will create the necessary directories.

File: Formula/nmh.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    (testpath/"Mail/inbox/1").write <<-EOM.undent
+      From: Mister Test <test@example.com>
+      To: Mister Nobody <nobody@example.com>
+      Date: Tue, 5 May 2015 12:00:00 -0000
+      Subject: Hello!
+
+      How are you?
+    EOM
+    ENV["TZ"] = "GMT"
+    output = `#{bin}/scan -width 80`
```
Comment (bfontaine) [inline Formula/nmh.rb]: Use `shell_output("…")` instead of `` `…` ``.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/nmh.rb (hunk 3 above)
Comment (kenh) [inline Formula/nmh.rb]: Just for my own edification; where is this documented? I freely admit I am not in any way a Ruby expert; I would just like to know what I need to read.

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/nmh.rb (hunk 3 above)
Comment (bfontaine) [inline Formula/nmh.rb]: Unfortunately I don’t think we documented this somewhere. It’s used in a lot of formulae (see e.g. [`aardvark_shell_utils.rb`](https://github.com/Homebrew/homebrew-core/blob/cb886e6d5b8255ad8139fce2a25b1afb9ea15589/Formula/aardvark_shell_utils.rb#L27-L29), [`cgrep.rb`](https://github.com/Homebrew/homebrew-core/blob/1668cbc72bf4f5d38e7316774251f266bfc2f8f7/Formula/cgrep.rb#L34-L39), [`ppss.rb`](https://github.com/Homebrew/homebrew-core/blob/2c5572c61df26d7915af8ee796bad3dfea66ed90/Formula/ppss.rb#L14)). This is a function [defined in Homebrew](https://github.com/Homebrew/brew/blob/028d155e97014076adf4e4dbc7c888fd871de321/Library/Homebrew/formula_assertions.rb#L27) that’s an equivalent of `` `…` `` but asserts that the command ran successfully.

(note to myself or anyone reading this: document `shell_output` somewhere)

**Review by bfontaine — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

File: Formula/nmh.rb (hunk 3 above)
Comment (MikeMcQuaid) [inline Formula/nmh.rb]: We need to improve our automated tests for this (e.g. `brew audit`) but thanks for the suggestion.

**Pull request closed without merging**